### PR TITLE
Update selectors and badge styles

### LINKED ISSUE #162: Change address dropdown/selector design
We should only style the current components (most importantly the `Autocomplete` and `Select`).
Currently we use `Autocomplete` in `AccountSelection`, `GenericAcconutSelection`, `MultiproxySelection` and `SignerSelection`
We can use the opportunity to refactor the code to use only 1 component as much as feasible in a separate issue. This issue centered around the design and it should be 1 dedicated PR.

https://www.figma.com/file/4dQC9kxf5uxCrAjpkzAK0D/Multix?type=design&node-id=229-12650&t=cYlBfIExNIk71DWG-0

New design
image

### PR DESCRIPTION
**Pull request by Lykhoyda**
closes #162

### CODE REVIEW

**Comment by Lykhoyda** [comment]
@Tbaut the styles of Select and Autocomplete are updated. There could be minor misalignment with the design. The short version where we display only the icon is done for "Select" component only. The PR is already big enough, any new updates and features would be better handled in a separate PR.

File: packages/ui/src/components/library/Autocomplete.tsx
Comment (Tbaut) [inline packages/ui/src/components/library/Autocomplete.tsx:19]: there are many `any`s here, can we import the MUIs types?

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (Tbaut) [inline packages/ui/src/components/selectors/GenericAccountSelection.tsx:140]: get we type this better?

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (Tbaut) [inline packages/ui/src/components/selectors/GenericAccountSelection.tsx:126]: Can we avoid this `any` please?

File: packages/ui/src/components/selectors/MultiProxySelection.tsx
Comment (Tbaut) [inline packages/ui/src/components/selectors/MultiProxySelection.tsx:74]: can we avoid this `any` please

File: packages/ui/src/components/EasySetup/BalancesTransfer.tsx
Comment (Tbaut) [inline packages/ui/src/components/EasySetup/BalancesTransfer.tsx:5]: If found a bug here when you have a moderately long list of accounts, it doesn't get displayed entirely and we need to scroll to actually show it all, where it should be above.

image

File: packages/ui/src/components/library/Autocomplete.tsx
Comment (Tbaut) [inline packages/ui/src/components/library/Autocomplete.tsx:81]: I find the icons way too small in a dropdown, and don't really display nicely with 2 information such as a name and address.
Identicons are a way to identify accounts quickly and need to be large enough for a selection I'd say. Ping @sweetpea22 

image

**Review by Tbaut — COMMENTED** [review]
Thanks a lot for this huge one. I made my best to review as thoroughly as possible. I'll ask @sweetpea22 to answer some of my comments here, and let's see how we move forward with these particular ones. There are so many changes, it's hard to review everything. Let me know what you want to fix in this PR or not, and we should create issues for those you'd prefer to tackle separately.

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (mpetrunic) [reply packages/ui/src/components/selectors/GenericAccountSelection.tsx:126]: PS. add linter rule to prevent it 😛

File: packages/ui/src/components/library/InputField.tsx
Comment (Lykhoyda) [reply packages/ui/src/components/library/InputField.tsx:54]: It's actually 0.5rem. Am I missing something?
Screenshot 2023-07-11 at 14 40 31

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (Lykhoyda) [reply packages/ui/src/components/selectors/GenericAccountSelection.tsx:126]: @mpetrunic for sure, but in the separate PR 😛

File: packages/ui/src/components/EasySetup/BalancesTransfer.tsx
Comment (Lykhoyda) [reply packages/ui/src/components/EasySetup/BalancesTransfer.tsx:5]: If you will start to scroll the content will move above the input. The behavior has changed due to the "disablePortal" props for Autocomplete. I disable it as it renders the models outside the "root" div and it was harder to customize the styles. 

Screenshot 2023-07-11 at 15 39 03

**Comment by Lykhoyda** [comment]
@Tbaut thank you for the detailed review. I updated the design according to the comments, please check again

File: packages/ui/src/components/library/InputField.tsx
Comment (Tbaut) [reply packages/ui/src/components/library/InputField.tsx:54]: I think Figma isn't consistent here then, and I didn't comment on the right component. The drop down menus have a 0.75 rem (the one I linked)
image

File: packages/ui/src/components/EasySetup/BalancesTransfer.tsx
Comment (Tbaut) [reply packages/ui/src/components/EasySetup/BalancesTransfer.tsx:5]: My list is very long, add some more accounts and you'll be able to reproduce. The positioning of the drop-down above or below is only related to the space available until the end of the screen, not to whether I scrolled or not. If the screen is high enough (btw I'm on a normal laptop, and it is) it's displayed below, and is cut.

File: packages/ui/src/components/library/InputField.tsx
Comment (Lykhoyda) [reply packages/ui/src/components/library/InputField.tsx:54]: @sweetpea22 can you please help us here? 😃 What is the expected border-radius to be implemented?

Comment (sweetpea22) [reply packages/ui/src/components/library/InputField.tsx:54]: it would be .75rem or 12px, sorry for the inconsistency. i will fix this in the board today

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (sweetpea22) [reply packages/ui/src/components/selectors/GenericAccountSelection.tsx]: hm yes i agree. i think making the gray a lot lighter will help. is there a preview url or should i run locally to try this out?

File: packages/ui/src/components/library/Autocomplete.tsx
Comment (sweetpea22) [reply packages/ui/src/components/library/Autocomplete.tsx:82]: yes it looks like my design here is wrong, it's too small to look good. I'm thinking to only use the big ones like this and remove the address in the dropdown for tidiness. but i'll give it a go to see how this looks first, brb
Screen Shot 2023-07-11 at 11 59 46 AM
Screen Shot 2023-07-11 at 12 00 11 PM

File: packages/ui/src/components/modals/Send.tsx
Comment (Tbaut) [reply packages/ui/src/components/modals/Send.tsx:277]: How about this comment?

File: packages/ui/src/components/selectors/MultiProxySelection.tsx
Comment (Tbaut) [reply packages/ui/src/components/selectors/MultiProxySelection.tsx:33]: This is still relevant

File: packages/ui/src/components/library/Autocomplete.tsx
Comment (Tbaut) [reply packages/ui/src/components/library/Autocomplete.tsx]: The `renderOption`, `onChange`, `isOptionEqualToValue` etc above still have some `any`.

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (Tbaut) [reply packages/ui/src/components/selectors/GenericAccountSelection.tsx]: There's a preview URL yes here for instance https://17fd413b.multix.pages.dev/

You can find them after a couple commits, click on the "View Deployment"
image

File: packages/ui/src/components/library/Autocomplete.tsx
Comment (Tbaut) [reply packages/ui/src/components/library/Autocomplete.tsx:82]: agreed, cool

**Comment by asnaith** [comment]
I took a look through all major screens after the recent changes and I did not notice any issues, nice visual improvements :)

**Review by asnaith — APPROVED** [review]

File: packages/ui/src/components/selectors/GenericAccountSelection.tsx
Comment (Tbaut) [reply packages/ui/src/components/selectors/GenericAccountSelection.tsx]: I discussed it with @sweetpea22 and we should go with the following, which is black on `gray[100]` like here https://www.figma.com/file/4dQC9kxf5uxCrAjpkzAK0D/Multix?node-id=502%3A14239&mode=dev
image

File: packages/ui/src/styles/theme.ts
Comment (Tbaut) [reply packages/ui/src/styles/theme.ts:153]: I discussed it with @sweetpea22 and we should go with the following, which is black on `gray[100]` like here https://www.figma.com/file/4dQC9kxf5uxCrAjpkzAK0D/Multix?node-id=502%3A14239&mode=dev
image

**Comment by Lykhoyda** [comment]
@Tbaut please take a look. I updated the types for the methods and fixed the long list appearance

**Review by Tbaut — COMMENTED** [review]
Thank you so much, this is looking great. I found a couple things that were forgotten, but I think we're good to go after that :)

File: packages/ui/src/components/library/Autocomplete.tsx
Comment (Tbaut) [reply packages/ui/src/components/library/Autocomplete.tsx:16]: The check icon is from the on-chain identity retrived automatically. Sorry I didn't mention it but you have to be on Kusama to see it.

Comment (Tbaut) [reply packages/ui/src/components/library/Autocomplete.tsx:16]: Also add it without giving it a name, that works. This is now displayed this way, which is unreadable. The name should be on 1 line and the address on another
image

**Comment by Tbaut** [comment]
There is another badge color that has been forgotten, and the background is still red, here: https://github.com/ChainSafe/Multix/blob/d5a92ccb04ea285c81a639aa870acbc5293b7652/packages/ui/src/components/Transactions/Transaction.tsx#L148-L149
image

File: packages/ui/src/components/select/SignerSelection.tsx
Comment (Tbaut) [reply packages/ui/src/components/select/SignerSelection.tsx:89]: > I would prefer that issue focus on one subject, otherwise, it's not enjoyable to work in one huge PR

Regarding this in particular, I totally agree. Please open an issue and we can tackle it in a dedicated PR.

The bug I mentioned here should however be fixed before we merge https://github.com/ChainSafe/Multix/pull/235#discussion_r1262680318

Comment (Lykhoyda) [reply packages/ui/src/components/select/SignerSelection.tsx:89]: @Tbaut the bug you mentioned is fixed

**Review by Tbaut — APPROVED** [review]
nice, that was yet another big one, thank you!

**Pull request merged**
